src/legacy/server/saved_objects/routes/export.ts

```diff
@@ -72,15 +77,24 @@ export const createExportRoute = (
     },
     async handler(request: ExportRequest, h: Hapi.ResponseToolkit) {
       const { savedObjectsClient } = request.pre;
-      const docsToExport = await getSortedObjectsForExport({
+      const exportStream = await getSortedObjectsForExport({
         savedObjectsClient,
         types: request.payload.type,
         objects: request.payload.objects,
         exportSizeLimit: server.config().get('savedObjects.maxImportExportSize'),
         includeReferencesDeep: request.payload.includeReferencesDeep,
       });
+
+      const docsToExport: string[] = await createPromiseFromStreams([
+        exportStream,
+        createMapStream((obj: unknown) => {
+          return stringify(obj);
+        }),
+        createConcatStream([]),
+      ]);
+
       return h
-        .response(docsToExport.map(doc => stringify(doc)).join('\n'))
+        .response(docsToExport.join('\n'))
```

Comment:
shouldn't we response with Stream? should reduce memory & CPU footprint as well

Reply:
I intentionally left this as a normal, non-streamed response. I didn't feel comfortable changing that as part of this PR. Maybe that's something we can investigate when we migrate these routes to the NP. Does the NP allow for streaming responses?

Reply:
yes https://github.com/elastic/kibana/pull/39767/files#diff-2cea1cfcff8d20213c7a4f71b2d9966cR81

Reply:
Awesome!